### PR TITLE
Init object and gcli packages from k/release

### PR DESCRIPTION
**Pull request by justaugustus**
Discussed on today's @kubernetes-sigs/release-engineering call and required for artifact promotion work: https://github.com/kubernetes/k8s.io/issues/2624

- object: Copy in k/release/pkg/object at 96c97b2f
- go.mod: Initialize module
- gcli: Copy in k/release/pkg/gcp/gcp.go at 96c97b2f
- object: Update import paths
- generated: Run `go mod tidy`

Signed-off-by: Stephen Augustus <foo@auggie.dev>

### CODE REVIEW

File: gcli/gcli.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2020 The Kubernetes Authors.
```
Comment (ameukam) [inline gcli/gcli.go:2]:
```suggestion
Copyright 2021 The Kubernetes Authors.
```

Comment (justaugustus) [reply gcli/gcli.go:2]: @ameukam -- This is a move, so let's maintain the copyright dates from the original code.

Comment (ameukam) [reply gcli/gcli.go:2]: @justaugustus 🤦🏾‍♂️

**Pull request merged**